src/lib/stores/UIStore.js

```diff
@@ -1,9 +1,31 @@
 import { observable, computed, action } from "mobx";
 import getConfig from "next/config";
 
+/**
+ * A mobx store for UI data
+ * @class AuthStore
+ */
+
 class UIStore {
+  /**
+   *`query` @observable
+   * @type @observable
+   * @param {boolean} cartOpen - Is the cart drawer open or closed
+   */
```

Comment:
`@param` is for function parameters and we probably want the type to indicate the actual type rather than `@observable`. Scaler values are actually not special observable values in MobX. It is the parent object that is technically the observable/reactive thing. So really this is just a normal boolean. Maybe format these like so:

```js
/**
 * Is the cart drawer open or closed
 *
 * @type Boolean
 * @default false
 */
```

Reply:
👍 Updated.

There is a mobx jsdoc plugin, I wasn't sure if that wsa comprehensive for what we'd be doing though.

https://www.npmjs.com/package/jsdoc-mobx-tags

Reply:
yeah, does not seem necessary for now